gpjax/mean_functions.py

```diff
@@ -25,9 +25,8 @@ def __call__(self, x: f64["N D"]) -> f64["N Q"]:
         """
         raise NotImplementedError
 
-    @property
     @abc.abstractmethod
-    def params(self) -> dict:
+    def _initialise_params(self, key: jnp.DeviceArray) -> Dict:
```

Comment:
We can use `PRNGKeyType` in future instead of a DeviceArray on the initialise methods as in #99 .